File: src/main/java/no/nav/syfo/unleash/ToggleImpl.java
package no.nav.syfo.unleash;

import no.finn.unleash.Unleash;
import no.finn.unleash.UnleashContext;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Primary;
import org.springframework.context.annotation.Profile;
import org.springframework.stereotype.Service;

@Service
@Profile("remote")
@Primary
public class ToggleImpl implements Toggle {
    private Unleash unleash;
    private boolean isProd;

    public ToggleImpl(Unleash unleash,
                      @Value("${fasit.environment.name:p}") String fasitEnvironmentName) {
        this.unleash = unleash;
        isProd = "p".equals(fasitEnvironmentName);
    }

    @Override
    public boolean isEnabled(FeatureToggle toggle, UnleashContext unleashContext) {
        if (isProd && !toggle.isAvailableInProd()) {
            return false;
        }
        return unleash.isEnabled(toggle.getToggleName(), unleashContext);
    }

    @Override
    public boolean isEnabled(FeatureToggle toggle) {
        if (isProd && !toggle.isAvailableInProd()) {
            return false;
        }
        return unleash.isEnabled(toggle.getToggleName());
    }

    public boolean isProd() {
        return isProd;
    }
}

File: src/main/kotlin/no/nav/syfo/config/unleash/ToggleImpl.kt
package no.nav.syfo.config.unleash

import no.finn.unleash.Unleash
import no.finn.unleash.UnleashContext
import org.springframework.beans.factory.annotation.Value
import org.springframework.context.annotation.Primary
import org.springframework.context.annotation.Profile
import org.springframework.stereotype.Service

@Service
@Profile("remote")
@Primary
class ToggleImpl(private val unleash: Unleash,
                 @Value("\${fasit.environment.name:p}") fasitEnvironmentName: String) : Toggle {
    override val isProd: Boolean = "p" == fasitEnvironmentName

    override fun isEnabled(toggle: FeatureToggle, unleashContext: UnleashContext?): Boolean =
            when {
                (isProd && !toggle.isAvailableInProd) -> false
                unleashContext != null -> unleash.isEnabled(toggle.toggleName, unleashContext)
                else -> unleash.isEnabled(toggle.toggleName)
            }
}
